### ISSUE TITLE
Game speed is generally slow

### ISSUE THREAD

**Issue by lynxlynxlynx** [ja2-stracciatella/ja2-stracciatella#99]
Originally reported by: **Anonymous**

---

As the title says, the game speed is slow in general, fo example the mercs moving slowly, screen scrolling, pressing the laptop button, etc. Most particularilly it can be noticed from the slow bullet flight speed; in vanilla JA2 it's lighting fast but in stracciatella it is very slow, someone else had this issue too. This slowness of gameplay appears to only be on gennady's new stracciatella versions since the game speed is the same in vanilla JA2 in Tron's original stracciatella releases, perhaps poorly optimized code? SDL issue maybe, have you updated it or some other component from Tron's original? If so perhaps my very old computer just doesn't handle the newer versions of them very well. Still a thing to look into IMO.

---
- Bitbucket: https://bitbucket.org/gennady/ja2-stracciatella/issue/99

**Comment by derSteFfi** [comment]
soo... the link is down, ja-galaxy moved in the meantime. i think this should be the new one:
http://thepit.ja-galaxy-forum.com/index.php?t=msg&goto=102405&&srch=Cool+Patch

The patch done there is trivial, added sleep cycles after each game loop, so no, i don't think this helps. The author even references ja2-stracciatella in the end as a 'cool' (temperaturewise) port. :)

There is a second idea presented in that thread that patches the game to consume a specific amount of time for one game cycle, but i think we already do this and it is controllable via MS_PER_GAME_CYCLE (see #148 ), right?

**Comment by lynxlynxlynx** [comment]
It was externalised, so it's even easier to test. Some side-by-side timing is required to see if 25 is the best possible default. All three bugs linked above test various aspects of that.

**Comment by kavol** [comment]
well, the user actions speeds seem to be okay, laptop opening roughly the same as I remember (if I remember anything at all after all those years :-)), but I have a great speed issue with enemy turns - it takes ages to resolve enemy movement ... if I see the enemy, it walks/shoots the same as the player but sometimes it looks just like it got stuck, taking e.g. 5 secs between shots from the same position, and if I don't see the enemy, sometimes it takes even over a minute before the progress bar gets to zero and sides switch; I don't observe such delays during the civilians' turn (except for Kingpin's "civilians" in San Mona, obviously)
this is pretty annoying, and I just don't understand that when my machine is Core i7-3520M, by orders of magnitude faster than what I had when playing the original ...

**Comment by lynxlynxlynx** [comment]
This just got fixed, so either use the experimental build or wait for the next release:
https://github.com/ja2-stracciatella/ja2-stracciatella/pull/488

hw doesn't matter.

**Comment by kavol** [comment]
LOL, I must have done 'git pull' just a few minutes before that, it could have saved me a lot of <s>time</s> game progress yesterday evening ... thanks!

**Comment by lynxlynxlynx** [comment]
I've never had any problems, so I'd just close this until someone finds a concrete and problematic difference somewhere.

**Comment by Manuel-K** [comment]
Yes. Close and keep the retest label.

**Comment by chmallon** [comment]
This problem depends on the used SDL backend.
SDL_UpdateRect() is *very* slow on some platforms, e.g. OS X.
RefreshScreen() called SDL_UpdateRect() multiple times in one frame.
The OP did not state the used platform, so it is hard to tell, whether this was his issue, but calling SDL_UpdateRect() multiple times per frame was definitely a mistake.

**Comment by hypertexthero** [comment]
I have managed to get Jagged Alliance 2 running with Stracciatella (an excellent application!) on an M1 Mac Mini running macOS 11.6 Big Sur, but I have the slow speed issue, particularly noticeable by slow-moving bullets. 

When ran JA2 on [macOS 10.13 High Sierra](https://hypertexthero.com/ja2-113-mac/) with the v1.13 patch I also encountered the slow speed issue, and fixed it using the [Fixing Flickering section of this post](http://www.furlo.sk/blog/odborny/2015-11-23-running_gog_jagged_alliance_2_on_mac_os_x_10.10.5#fixing_flickering). Unfortunately the instructions don’t work in my current setup because I do not see the `ddr=gdi` option under Settings.

Any suggestions to get the game running quickly again on macOS 11.6 Big Sur on an M1 are very much appreciated.
